tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2608,7 +2645,39 @@ private String handleDateFormatItem(String xpath, String value, boolean showCont
                 DateFormatSymbols dfs = sdf.getDateFormatSymbols();
                 dfs.setTimeSeparatorString(timeSeparator);
                 sdf.setDateFormatSymbols(dfs);
-                if (id == null || id.indexOf('B') < 0) {
+                if (parts.contains("availableFormats")) {
+                    String example = addExampleResult(sdf.format(DATE_SAMPLE) + ", " + sdf.format(DATE_SAMPLE5), "", showContexts);
+                    String altId = generateAltPattern(id);
+                    if (!altId.equals(id)) {
+                        String altXpath = "//ldml/dates/calendars/calendar[@type=\""
+                            + calendar
+                            + "\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\""
+                            + altId
+                            + "\"]";
+                        String forcedAltValue = cldrFile.getWinningValue(altXpath);
+                        String unforcedAltValue = generateAltPattern(value);
+                        String forcePrefix = forcedAltValue != null ? "" : EXAMPLE_OF_INCORRECT;
+                        
+                        
+                        sdf =
+                            icuServiceBuilder.getDateFormat(calendar, unforcedAltValue, numbersOverride);
+                        sdf.setTimeZone(ZONE_SAMPLE);
+                        defaultNumberingSystem =
+                                cldrFile.getWinningValue("//ldml/numbers/defaultNumberingSystem");
+                        timeSeparator =
+                                cldrFile.getWinningValue(
+                                        "//ldml/numbers/symbols[@numberSystem='"
+                                                + defaultNumberingSystem
+                                                + "']/timeSeparator");
+                        dfs = sdf.getDateFormatSymbols();
+                        dfs.setTimeSeparatorString(timeSeparator);
+                        sdf.setDateFormatSymbols(dfs);
```

Comment:
I copy-pasted this chunk from above. The only change is `icuServiceBuilder.getDateFormat(calendar, unforcedAltValue, numbersOverride);` instead of `icuServiceBuilder.getDateFormat(calendar, value, numbersOverride);` Is there a way to just change the pattern on a SimpleDateFormat instead of creating a new one?

Reply:
You could pull that chunk of code out into a function, and call with slightly different parameters. Better that code copying (which is harder to maintain).